src/array/struct_.rs

```diff
@@ -8,6 +8,7 @@ use crate::{
 };
 
 use super::{ffi::ToFfi, new_empty_array, new_null_array, Array, FromFfi};
+use crate::array::ArrayRef;
```

Comment:
can we use `Arc<dyn Array>`? `ArrayRef` is only kept in the public API for compatiblity with DataFusion. For people not used to it, this adds an extra layer of indirection in reading code that is not necessary.